pubsub/google/cloud/pubsub_v1/subscriber/policy/thread.py

```diff
@@ -153,10 +158,16 @@ def on_exception(self, exception):
 
         This will cause the stream to exit loudly.
         """
-        # If this is DEADLINE_EXCEEDED, then we want to retry.
-        # That entails just returning None.
-        deadline_exceeded = grpc.StatusCode.DEADLINE_EXCEEDED
-        if getattr(exception, 'code', lambda: None)() == deadline_exceeded:
+        if isinstance(exception, exceptions.GoogleAPICallError):
+            code = exception.grpc_status_code
+        else:
+            code = getattr(exception, 'code', None)
+            if callable(code):
```

Comment:
???

Reply:
`_Rendevous.code` is callable, which is why he did that. In theory we should catch all of them, but if we miss one...

Reply:
But it's is a `grpc.Call` then `code` will *always* be callable, what's the case where there's a `code` attribute?

Reply:
I'm just going to do an `isinstance` check for `_Rendezvous`.